### PR TITLE
Add table_aws_route53_domain. closes #59

### LINKED ISSUE #59: Add table_aws_route53_domain

### PR DESCRIPTION
**Pull request by Priyanka585464**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  domain_name,
  auto_renew,
  expiry
from
  aws_route53_domains;
```
```
+------------------+------------+---------------------+
| domain_name             | auto_renew | expiry              |
+------------------+------------+---------------------+
| priyankatest.com | true       | 2022-04-29 09:36:51 |
+------------------+------------+---------------------+
```
### List Route53 domain names are renewed before their expiration.

```sql
select
  domain_name,
  auto_renew,
  expiry
from
  aws_route53_domains
where
  auto_renew = 'true';
```
```
+------------------+------------+---------------------+
| domain_name             | auto_renew | expiry              |
+------------------+------------+---------------------+
| priyankatest.com | true       | 2022-04-29 09:36:51 |
+------------------+------------+---------------------+
```

### List Route53 domain names are renewed before Expiry 7 Days

```sql
select
  domain_name,
  auto_renew,
  expiry
from
  aws_route53_domains
where
  expiry <= (expiry - interval '7' day);
```
```
+------+------------+--------+
| domain_name | auto_renew | expiry |
+------+------------+--------+
+------+------------+--------+
```

### Check your domain names have the Transfer Lock feature enabled in order to keep them secure.

```sql
select
  domain_name,
  expiry,
  transfer_lock
from
  aws_route53_domains
where
  transfer_lock = 'true';
```
```
+------------------+---------------------+
| domain_name             | expiry              |
+------------------+---------------------+
| priyankatest.com | 2022-04-29 09:36:51 |
+------------------+---------------------+
```

### CODE REVIEW

File: aws/table_aws_route53_domains.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "transfer_lock",
+				Description: "Indicates whether a domain is locked from unauthorized transfer to another party.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the parameter.",
```
Comment (Subhajit97) [inline aws/table_aws_route53_domains.go:48]:
```suggestion
				Description: "A list of tags assigned to the resource.",
```

File: aws/service.go [hunk 2]
```diff
@@ -844,6 +845,27 @@ func RedshiftService(ctx context.Context, d *plugin.QueryData, region string) (*
 	return svc, nil
 }
 
+// Route53DomainsService returns the service connection for AWS route53 domains service
+func Route53DomainsService(ctx context.Context, d *plugin.QueryData, region string) (*route53domains.Route53Domains, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed Route53Domains")
+	}
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("route53domain-%s", region)
```
Comment (Subhajit97) [inline aws/service.go:854]: Is the service regional?

File: aws/table_aws_route53_domains.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	svc, err := Route53DomainsService(ctx, d, "us-east-1")
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53domains.ListDomainsInput{}
+
+	// Get call
+	data, err := svc.ListDomains(params)
```
Comment (Subhajit97) [inline aws/table_aws_route53_domains.go:117]: Can we use `GetDomainDetail` API for GET?
https://docs.aws.amazon.com/Route53/latest/APIReference/API_domains_GetDomainDetail.html

File: aws/table_aws_route53_domains.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+	logger.Trace("getRoute53Domain")
+
+	// Create session
+	svc, err := Route53DomainsService(ctx, d, "us-east-1")
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53domains.ListDomainsInput{}
```
Comment (Subhajit97) [inline aws/table_aws_route53_domains.go:114]: Don't we need any filter for GET call?

File: aws/table_aws_route53_domains.go [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+func getAwsRoute53DomainAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsRoute53DomainAkas")
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	commonColumnData := c.(*awsCommonColumnData)
+	aka := "arn:" + commonColumnData.Partition + ":route53domains:" + ":" + commonColumnData.AccountId
```
Comment (Subhajit97) [inline aws/table_aws_route53_domains.go:159]: Can we also include `domainName` in ARN to make it unique?

File: aws/table_aws_route53_domains.go [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+	return []string{aka}, nil
+}
+
+//// TRANSFORM FUNCTION
```
Comment (Subhajit97) [inline aws/table_aws_route53_domains.go:163]:
```suggestion
//// TRANSFORM FUNCTIONS
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@Priyanka585464 I have added some suggestions. Please see comments, thanks!

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Can we have DNSSEC setting from this table ?
https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/domain-configure-dnssec.html

**Comment by Subhajit97** [comment]
> Can we have DNSSEC setting from this table ?
> https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/domain-configure-dnssec.html


@rajmohanty17 , currently API doesn't return any details about DNSSEC configuration. There is a property that indicated the DNSSEC configuration (https://docs.aws.amazon.com/Route53/latest/APIReference/API_domains_GetDomainDetail.html#Route53Domains-domains_GetDomainDetail-response-DnsSec), but it always return `null`.

File: aws/plugin.go [hunk 7]
```diff
@@ -28,6 +28,160 @@ func Plugin(ctx context.Context) *plugin.Plugin {
...
+			"aws_vpc_security_group_rule":            tableAwsVpcSecurityGroupRule(ctx),
+			"aws_vpc_subnet":                         tableAwsVpcSubnet(ctx),
+			"aws_vpc_vpn_gateway":                    tableAwsVpcVpnGateway(ctx),
+			"aws_waf_rule":                           tableAwsWAFRule(ctx),
+			"aws_wafv2_ip_set":                       tableAwsWafv2IpSet(ctx),
+			"aws_wafv2_regex_pattern_set":            tableAwsWafv2RegexPatternSet(ctx),
+			"aws_wafv2_rule_group":                   tableAwsWafv2RuleGroup(ctx),
+			"aws_wafv2_web_acl":                      tableAwsWafv2WebAcl(ctx),
+			"aws_wellarchitected_workload":           tableAwsWellArchitectedWorkload(ctx),
+=======
```
Comment (Subhajit97) [inline aws/plugin.go:184]: @Priyanka585464 I think file is not properly rebased. Please verify again.
```suggestion
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@Priyanka585464 Please rebase the branch.

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**